Automattic/wp-calypso
# Checkout: Do not display contact details form if the cart is only renewals, take 2

#### Changes proposed in this Pull Request

This is another attempt at https://github.com/Automattic/wp-calypso/pull/44468, which was reverted in https://github.com/Automattic/wp-calypso/pull/44506

This PR changes the behavior of the new checkout form's contact details so that it will not display if the cart contains only renewal items and the existing cached contact details are valid. This behavior should mostly match [old checkout](https://github.com/Automattic/wp-calypso/blob/096d4759a964ebb19bcf0a068e51d438e9fab080/client/my-sites/checkout/checkout/index.jsx#L917). Previously, we had been displaying the contact details form fields any time that there were domain products in the cart.

In addition, it prevents sending the `domain_details` property to the transactions endpoint (and to paypal, etc.) when the cart contains only renewals.

#### Screenshots

Before:

![Screen Shot 2020-07-27 at 2 21 16 PM](https://user-images.githubusercontent.com/2036909/88577259-bb636c80-d014-11ea-965e-c9ee82fbc4da.png)

After:

![Screen Shot 2020-07-27 at 2 17 32 PM](https://user-images.githubusercontent.com/2036909/88577272-be5e5d00-d014-11ea-92dc-fe44dd927652.png)


#### Testing instructions

- Add a domain product to your cart (we may want to test this with different domain products for completeness) and visit checkout.
- Verify that you see the full "Contact information" step including name, address, and phone number.
- Complete the purchase. Verify that the purchase is successful. (Bonus: monitor the `/me/transactions` API call and make sure that `domain_details` is sent with the full contact data.)
- Visit http://calypso.localhost:3000/me/purchases and click the domain, then click "Renew now". You'll be redirected to checkout.
- Verify that the "Contact information" step only includes the "tax" information which is just postal code and country.
- Complete the purchase. Verify that the purchase is successful. (Bonus: monitor the `/me/transactions` API call and make sure that `domain_details` is sent with `null`.)
